packages/core-sdk/src/utils/registerHelper.ts

```diff
@@ -0,0 +1,1319 @@
+    rpcClient,
+    wallet,
+  );
+  const licenseAttachmentWorkflowsClient = new LicenseAttachmentWorkflowsClient(rpcClient, wallet);
+  const derivativeWorkflowsClient = new DerivativeWorkflowsClient(rpcClient, wallet);
+
+  if ("licenseTermsData" in request) {
+    const { licenseTermsData } = await validateLicenseTermsData(
+      request.licenseTermsData,
+      rpcClient,
+    );
+    const requestWithTerms = { ...baseRequest, licenseTermsData };
+
+    if ("royaltyShares" in request) {
+      return await transferRegisterIpAndAttachPilTermsAndDeployRoyaltyVaultRequest({
+        request: requestWithTerms,
+        royaltyTokenDistributionWorkflowsClient,
+        chainId,
+        wallet,
+        calculatedDeadline,
+        ipIdAddress,
+        royaltyShares: request.royaltyShares,
+      });
+    }
+
+    return await transferRegisterIpAndAttachPilTermsRequest({
+      request: requestWithTerms,
+      licenseAttachmentWorkflowsClient,
+      calculatedDeadline,
+      ipIdAddress,
+      wallet,
+      chainId,
+    });
+  }
+
+  if ("derivData" in request) {
+    const derivData = await validateDerivativeData({
+      derivativeDataInput: request.derivData,
+      rpcClient,
+      chainId,
+      wallet,
+    });
+    const totalFees = await calculateDerivativeMintingFee({
+      derivData,
+      rpcClient,
+      chainId,
+      wallet,
+    });
+    if ("royaltyShares" in request) {
+      return await transferRegisterIpAndMakeDerivativeAndDeployRoyaltyVaultRequest({
+        request: {
+          ...baseRequest,
+          derivData,
+        },
+        calculatedDeadline,
+        ipIdAddress,
+        wallet,
+        chainId,
+        royaltyTokenDistributionWorkflowsClient,
+        totalFees,
+        royaltyShares: request.royaltyShares,
+      });
+    }
+
+    return await transferRegisterDerivativeIpRequest({
+      request: {
+        ...baseRequest,
+        derivData,
+      },
+      calculatedDeadline,
+      ipIdAddress,
+      wallet,
+      chainId,
+      derivativeWorkflowsClient,
+      totalFees,
+    });
+  }
+
+  throw new Error("Invalid register request type");
+};
+
+export const transformRegistrationRequest = async <
+  T extends TransformIpRegistrationWorkflowRequest,
+>({
+  request,
+  rpcClient,
+  wallet,
+  chainId,
+}: TransformRegistrationRequestConfig): Promise<TransformIpRegistrationWorkflowResponse<T>> => {
+  if ("spgNftContract" in request) {
+    return handleMintAndRegisterRequest<T>({ request, rpcClient, wallet, chainId });
+  } else if ("nftContract" in request && "tokenId" in request) {
+    return handleRegisterRequest<T>({ request, rpcClient, wallet, chainId });
+  } else if ("ipRoyaltyVault" in request) {
+    return handleDistributeRoyaltyTokensRequest<T>({ request, rpcClient, wallet, chainId });
+  }
+
+  throw new Error("Invalid registration request type");
+};
+
+const aggregateTransformIpRegistrationWorkflow = (
```

Comment:
lets add some comments to some of these helper functions. I think its not clear from the name exactly what their responsibilities are.